x/crosschain/keeper/cctx.go

```diff
@@ -81,6 +81,23 @@ func (k Keeper) GetCrossChainTx(ctx sdk.Context, index string) (val types.CrossC
 	return val, true
 }
 
+// GetCrossChainTxError returns the error message for a given cctx index.
+func (k Keeper) GetCrossChainTxError(ctx sdk.Context, index string) (errMsg string, found bool) {
```

Comment:
Why implementing this function? It is not used and I don't think we will need it as you can fetch the object and read the error message from it.

In general we should keep message that interact with the store as minimal as possible for simplicity

Reply:
Deleted.